tests/test_ner4opt.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
```

Comment:
1) not a big fan of matching the output of binary pickle files. How big these pickles would be if we add as dictionary objects inside the tests? OR as files into under tests/data

2) can we add some simple unit tests, with decent problem descriptions against hard-coded output list of dictionaries.. so that we can "see" what is matched. 

3) add a test_usage_example() that is "exactly" the same as the one in readme and assert output

Reply:
overall, tests seem quite weak. and ner4opt.utils seem not tested at all. can we add at least a few tests for utilities?

Reply:
I added more tests and currently the coverage should be close to 99%. I added usage tests for all model types, instead of pickled fixtures, which seem blackboxy, I changed it to json. Reasoning for adding it that way is to make sure, each model output for our known problems is exact and complete, which helps with reproducibility whenever changes are made to main library code. Hope that works. I also added tests for invalid usage, utils and features.